Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39027&attempt_id=ff15f1b7-4bd0-46d1-9a1f-1fea0631de50)

Attempted: grass

<details><summary>Partial log (click to expand)</summary><p>

```
       ^~~~~
/build/hdf-4.2.13/hdf/src/hdatainfo.h:53:32: error: unknown type name 'int32'; did you mean 'int'?
   (int32 vgid, intn attrindex, int32 *offset, int32 *length);
                                ^~~~~
                                int
building of '/nix/store/i5jnsjnjf02gn92g3k89z9khb58wg2s4-hdf-4.2.13.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/bj7rrq4w4sdb2as9gc6wcnr60050ds9d-gdal-2.2.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/x71ddji531i33kjlc197qr1wbsajnlzq-libLAS-1.8.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1bkd824q4kn48g6rlx651avsq50v6g9h-grass-7.2.2.drv': 2 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/1bkd824q4kn48g6rlx651avsq50v6g9h-grass-7.2.2.drv' failed
```
</p></details>

--- a/pkgs/applications/gis/grass/default.nix
+++ b/pkgs/applications/gis/grass/default.nix
@@ -1,6 +1,6 @@
 { stdenv, fetchurl, flex, bison, pkgconfig, zlib, libtiff, libpng, fftw
 , cairo, readline, ffmpeg, makeWrapper, wxGTK30, netcdf, blas
-, proj, gdal, geos, sqlite, postgresql, mysql, python2Packages
+, proj, gdal, geos, sqlite, postgresql, mysql, python2Packages, libLAS
 }:
 
 stdenv.mkDerivation {
@@ -12,7 +12,8 @@ stdenv.mkDerivation {
 
   nativeBuildInputs = [ pkgconfig ];
   buildInputs = [ flex bison zlib proj gdal libtiff libpng fftw sqlite cairo
-  readline ffmpeg makeWrapper wxGTK30 netcdf geos postgresql mysql.connector-c blas ]
+  readline ffmpeg makeWrapper wxGTK30 netcdf geos postgresql mysql.connector-c blas
+  libLAS ]
     ++ (with python2Packages; [ python dateutil wxPython30 numpy ]);
 
   # On Darwin the installer tries to symlink the help files into a system
@@ -33,6 +34,7 @@ stdenv.mkDerivation {
     "--with-mysql-includes=${mysql.connector-c}/include/mysql"
     "--with-mysql-libs=${mysql.connector-c}/lib/mysql"
     "--with-blas"
+    "--with-liblas=${libLAS}/bin/liblas-config"
   ];
 
   # Otherwise a very confusing "Can't load GDAL library" error

--- a/pkgs/development/libraries/libLAS/default.nix
+++ b/pkgs/development/libraries/libLAS/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, boost, cmake, gdal, libgeotiff, libtiff, LASzip }:
+{ stdenv, fetchurl, boost, cmake, gdal, libgeotiff, libtiff, LASzip, fixDarwinDylibNames }:
 
 stdenv.mkDerivation rec {
   name = "libLAS-1.8.1";
@@ -9,14 +9,22 @@ stdenv.mkDerivation rec {
     sha256 = "0xjfxb3ydvr2258ji3spzyf81g9caap19ql2pk91wiivqsc4mnws";
   };
 
-  buildInputs = [ boost cmake gdal libgeotiff libtiff LASzip ];
+  buildInputs = [ boost cmake gdal libgeotiff libtiff LASzip ]
+                ++ stdenv.lib.optional stdenv.isDarwin fixDarwinDylibNames;
 
+  cmakeFlags = [
+    "-DGDAL_CONFIG=${gdal}/bin/gdal-config"
+  ];
+
+  postFixup = stdenv.lib.optionalString stdenv.isDarwin ''
+    install_name_tool -change "@rpath/liblas.3.dylib" "$out/lib/liblas.3.dylib" $out/lib/liblas_c.dylib
+  '';
 
   meta = {
     description = "LAS 1.0/1.1/1.2 ASPRS LiDAR data translation toolset";
     homepage = http://www.liblas.org;
     license = stdenv.lib.licenses.bsd3;
-    platforms = stdenv.lib.platforms.linux;
+    platforms = stdenv.lib.platforms.unix;
     maintainers = [ stdenv.lib.maintainers.michelk ];
   };
 }

